### PR TITLE
Add Integration Tests for HTTP and WebSocket APIs

### PR DESCRIPTION
**Pull request by believemanasseh**
- Enabled cross-origin resource sharing (CORS) for frontend app running on localhost
- Wrote fixtures and test functions for HTTP and WebSocket API
- Created skeleton implementation of room creation event handler
- Formatted code with black

### CODE REVIEW

File: backend/tests/test_events.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    data = {
+        "name": "Test room",
+        "description": "Test description",
+        "project": "https://project-link.com",
+    }
+    socket_client.emit("create_room", json.dumps(data))
+    response = list(
+        filter(lambda dict: dict["name"] == "create_room", socket_client.get_received())
+    )[0]
+    assert "id" in json.loads(response["args"][0])
```
Comment (jaybuurdd) [inline backend/tests/test_events.py:16]: This test case failed.

image

It was expecting `id` so something is wrong with the response if you expect that to be returned.

File: backend/app.py [hunk 2]
```diff
@@ -1,35 +1,47 @@
...
     _app.logger.info(f"Websockets registered")
-    
-def create_app():
+
+
+def create_app(testing=False):
     app = Flask(__name__)
-    app.config.from_object(get_config())
+    CORS(
+        app, origins=["http://localhost:15173"], supports_credentials=True
```
Comment (ZakariaTalhami) [inline backend/app.py]: Do we want to have this fixed to `"http://localhost:15173"`? wouldnt it be better if it was set dynamically by the environment variables?

File: backend/services/utils.py [hunk 3]
```diff
@@ -1,25 +1,34 @@
...
         _app.logger.info("Start websocket config")
         socketio = SocketIO(_app, cors_allowed_origins="*")
         _app.logger.info(f"End websocket config {socketio}")
     except Exception as e:
-        _app.logger.error(f"Error during websocket config: {e}")     
\ No newline at end of file
+        _app.logger.error(f"Error during websocket config: {e}")
+
+
+def generate_invite_token() -> str:
```
Comment (ZakariaTalhami) [inline backend/services/utils.py:33]: What would be the purpose of this function?

**Review by jaybuurdd — APPROVED** [review]

File: backend/services/utils.py (hunk 3 above)
Comment (believemanasseh) [reply backend/services/utils.py:33]: For generating the invite token associated with a room session.

File: backend/app.py (hunk 2 above)
Comment (believemanasseh) [reply backend/app.py]: I just pushed a fix for that. You can review my latest commit

File: backend/tests/test_events.py (hunk 1 above)
Comment (believemanasseh) [reply backend/tests/test_events.py:16]: Yes, basically the feature being tested at the time of writing this test hadn't been properly implemented, so it's expected to fail.

File: backend/app.py [hunk 4]
```diff
@@ -1,35 +1,48 @@
...
     app = Flask(__name__)
-    app.config.from_object(get_config())
+
+    if testing:
+        app.config.from_object(get_testing_config())
+    else:
+        app.config.from_object(get_config())
+
+    origins = resolve_origins(app.config["CORS_ALLOWED_ORIGINS"])
+    CORS(app, origins=origins, supports_credentials=True)  # Enable CORS
```
Comment (ZakariaTalhami) [inline backend/app.py:41]: Nice!

**Review by ZakariaTalhami — APPROVED** [review]
+1 from me.

**Review by borhara — APPROVED** [review]
LGTM

**Pull request merged**